File: 2236-MaximumTwinSumOfALinkedList/2236-MaximumTwinSumOfALinkedList.js
// Last updated: 4/29/2026, 10:21:32 AM
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {number}
 */
var pairSum = function(head) {
    const arr = [];
    let node = head;
    while(node) {
        arr.push(node);
        node = node.next;
    }
    const n = arr.length;
    let max;
    for(let i = 0; i < n / 2; i++) {
        const twinSum = arr[i].val + arr[n-1-i].val;
        if(max === undefined || twinSum > max) {
            max = twinSum;
        }
    }
    return max;
};

File: 2236-MaximumTwinSumOfALinkedList/2236-MaximumTwinSumOfALinkedList.py
# Last updated: 4/29/2026, 10:21:37 AM
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
# 3:39s
# O(n)
# O(n)
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        node = head
        l = []
        while node != None:
            l.append(node)
            node = node.next
        n = len(l)
        max = None
        for i in range(n):
            twinIdx = n - 1 - i
            sum = l[i].val + l[twinIdx].val
            if max == None:
                max = sum
            elif max < sum:
                max = sum
        return max